s2spy/_resample.py

```diff
@@ -222,21 +241,28 @@ def resample(
         raise ValueError("Generate a calendar map before calling resample")
 
     # get i_interval to update resampled data
-    i_interval = mapped_calendar._rename_i_intervals(intervals).columns.values #pylint: disable=protected-access
+    # pylint: disable=protected-access
+    i_interval = mapped_calendar._rename_i_intervals(
+        intervals
+    ).columns.values
 
     utils.check_timeseries(input_data)
     # This check is still valid for all calendars with `freq`, but not for CustomCalendar
     # TO DO: add this check when all calendars are rebased on the CustomCalendar
-    #utils.check_input_frequency(mapped_calendar, input_data)
+    # utils.check_input_frequency(mapped_calendar, input_data)
 
     if isinstance(input_data, PandasData):
         resampled_data = resample_pandas(mapped_calendar, input_data)
         # update i_interval based on the calendar
         resampled_data.i_interval = np.tile(i_interval[::-1], len(intervals.index))
     else:
+        if isinstance(input_data, xr.DataArray):
+            input_data.name = "data" if input_data.name is None else input_data.name
+            input_data = input_data.to_dataset()
```

Comment:
I have renamed the function to `resample_dataset` to properly reflect its functionality. With the refactoring of `resample` that function has also become quite a bit emptier.
Can you have another look to see if it is OK now? Otherwise I can add a wrapper function with the `resample_xarray` name that does the input data checks and conversions.